src/desc.rs

```diff
@@ -5,32 +5,76 @@ use std::collections::{BTreeSet, HashMap};
 use std::hash::Hasher;
 
 use fnv::FnvHasher;
-use regex::Regex;
 
 use crate::errors::{Error, Result};
 use crate::metrics::SEPARATOR_BYTE;
 use crate::proto::LabelPair;
 
-// Details of required format are at
-// https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
-fn is_valid_metric_name(name: &str) -> bool {
-    lazy_static! {
-        static ref VALIDATOR: Regex =
-            Regex::new("^[a-zA-Z_:][a-zA-Z0-9_:]*$").expect("Regex to be valid.");
+#[cfg(not(feature = "regex"))]
+mod validation {
+    fn matches_charset_without_colon(c: char) -> bool {
+        c.is_ascii_alphabetic() || c == '_'
     }
 
-    VALIDATOR.is_match(name)
+    fn matches_charset_with_colon(c: char) -> bool {
+        matches_charset_without_colon(c) || c == ':'
+    }
+
+    // check if there's at least one char
+    // the charset_validator returns true on zeroth char
+    // the charset_validator returns true on all remaining chars or they are digits if it returned
+    // false
+    // Equivalent to regex ^[?][?0-9]*$ where ? denotes char set as validated by charset_validator
```

Comment:
```suggestion
    /// Equivalent to regex `^[?][?0-9]*$` where `?` denotes char set as validated by `charset_validator`.
```

I think the above would suffice here.